File: examples/simple_trixi_controller.jl
using LibTrixi

project_dir = "libtrixi-julia"
libelixir = "LibTrixi.jl/examples/libelixir_tree1d_dgsem_advection_basic.jl"

println("*** Trixi controller ***   Set up Trixi simulation")
handle = trixi_initialize_simulation(libelixir)

println("*** Trixi controller ***   Current time step length: ", trixi_calculate_dt(handle))

println("*** Trixi controller ***   Entering main loop")

while trixi_is_finished(handle) == 0
    trixi_step(handle)
end

println("*** Trixi controller ***   Finalize Trixi simulation")
handle = trixi_finalize_simulation(handle)

File: examples/simple_trixi_controller.f90
program simple_trixi_controller_f
  use LibTrixi
  use, intrinsic :: iso_fortran_env, only: error_unit
  use, intrinsic :: iso_c_binding, only: c_int, c_null_char

  implicit none

  integer(c_int) :: handle
  character(len=256) :: argument

  if (command_argument_count() < 1) then
    call get_command_argument(0, argument)
    write(error_unit, '(a)') "ERROR: missing arguments: PROJECT_DIR LIBELIXIR_PATH"
    write(error_unit, '(a)') ""
    write(error_unit, '(3a)') "usage: ", trim(argument), " PROJECT_DIR LIBELIXIR_PATH"
    stop 1
  else if (command_argument_count() < 2) then
    call get_command_argument(0, argument)
    write(error_unit, '(a)') "ERROR: missing argument: LIBELIXIR_PATH"
    write(error_unit, '(a)') ""
    write(error_unit, '(3a)') "usage: ", trim(argument), " PROJECT_DIR LIBELIXIR_PATH"
    stop 1
  end if

  ! Initialize Trixi
  call get_command_argument(1, argument)
  call trixi_initialize(argument)

  ! Print version information
  write(*, '(a, a)') "libtrixi version: ", trixi_version()

  ! Set up the Trixi simulation
  ! We get a handle to use subsequently
  call get_command_argument(2, argument)
  handle = trixi_initialize_simulation(argument)

  ! Get time step length
  write(*, '(a, e14.8)') "Current time step length: ", trixi_calculate_dt(handle)

  ! Main loop
  do
    ! Exit loop once simulation is completed
    if ( trixi_is_finished(handle) ) exit

    call trixi_step(handle)
  end do

  ! Finalize Trixi simulation
  call trixi_finalize_simulation(handle)

  ! Finalize Trixi
  call trixi_finalize()
end program
